ls1intum/Artemis
# `Development`: Refactor InvalidPasswordException to make it unambiguous

### Checklist
#### General


#### Server


### Motivation and Context

The naming of the exception was considered a bit ambiguous since the exception should be thrown if the provided password did not match the requirements (registration, etc). Instead it should point to the requirements that were being violated in the registration process. This was described in the issue #4628 available on https://github.com/ls1intum/Artemis/issues/4628. 

### Description

I refactored the name of the exception (in the original occurrences pointed in the issue #4508 (available on https://github.com/ls1intum/Artemis/pull/4508#discussion_r787867568) and other that also used the same exception name. I used the PasswordViolatesRequirementsException instead of the proposed InvalidFormPasswordException. The test has been run in Bamboo (https://bamboobruegge.in.tum.de/browse/ARTEMIS-TESTS3197-1) and it has built normally. I'm not sure how the other tests described above would apply.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Programming Exercise with Complaints enabled

1. Log in to Artemis
2. Navigate to Course Administration
3. Try to register with a password that does not meet the requirements set there and see if the exception gets thrown. 
4. The test should be carried for the following scenarios:

- Registration (Where activated)
- User Creation (Admin)
- User Edit (Admin)
- Password reset (Internal user only)
- Password change (Internal user only)

### Review Progress

#### Code Review
#### Manual Tests
